models/Platform.ts

```
@@ -6,7 +8,13 @@
  * - `wpcom`: The experiment is being run on the back-end, like APIs which are
  *   usually written in PHP. Email and landing pages experiments are likely run here.
  */
-export enum Platform {
+enum Platform {
   Calypso = 'calypso',
   Wpcom = 'wpcom',
 }
+
+function toPlatform(input: string) {
+  return Platform[capitalize(input) as keyof typeof Platform] || Platform.Calypso
```

Defaulting to `Calypso` here and to `Staging` in `toStatus()` may mask bugs. Let's throw exceptions for bad strings.

---

Actually, given that the enum is defined using strings, it turns out this function is unnecessary. Can simply use `input as Platform`.

---

Yeah, I meant to ask you about the defaulting of the Platform and Status. I was only doing that because we have `status` and `platform` defined as not being nullable in the experiments interfaces. So, TS was complaining about assigning `Status | undefined` to `Status` (similarly for `Platform`).

> Actually, given that the enum is defined using strings, it turns out this function is unnecessary. Can simply use input as Platform.

I tried this. It compiles, but it results in 'Calypso' instead of 'calypso'.

---

> I tried this. It compiles, but it results in 'Calypso' instead of 'calypso'.

Are you sure? I ran it inlined with the constructor from https://github.com/Automattic/abacus/pull/66#discussion_r425508873 and I'm seeing the lowercase version in the table.

---

I am not sure what I saw, but the updated unit tests are passing with what I we are expecting. I am curious what TS is doing behind the scenes.